test_ros2cli/test/config_ros2action_test.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+
+from launch.substitutions import LaunchConfiguration
+
+from launch_ros.actions import Node
+
+sys.path.append(os.path.dirname(__file__))
+
+from test_config import TestConfig  # noqa
+
+
+config = TestConfig()
```

Comment:
To clarify, what I was proposing was to couple the command, arguments, actions, output (msgs) close together into a single object. Then we don't need this notion of option key. E.g.

```python
config = [
    TestConfig(
        command='action',
        arguments=['info', '/fibonacci'],
        actions=[get_action_server_node_action()],
        expected_output=common_info_output + ['/fibonacci_action_server']
    ),
    TestConfig(
        command='action',
        arguments=['info', '-t', '/fibonacci'],
        actions=[get_action_server_node_action()],
        expected_output=common_info_output + ['/fibonacci_action_server [action_tutorials/action/Fibonacci]']
    ),
    ...,
]
```

IMO, it makes the tests more readable, especially when adding/removing/modifying tests.

Reply:
We can also default the last three kwargs to `[]`.

Reply:
I'm not the best way to parameterize the test on this list structure though. Maybe we could still use a dictionary where the key is the test description?